File: android/app/src/main/java/com/windchaser/runningos/domain/repository/RunningRepository.kt
package com.windchaser.runningos.domain.repository

import com.windchaser.runningos.domain.model.RunningActivity
import com.windchaser.runningos.domain.model.Route
import com.windchaser.runningos.domain.model.UserStats
import kotlinx.coroutines.flow.Flow

/**
 * 跑步数据 Repository 接口
 */
interface RunningRepository {
    /**
     * 获取用户统计数据
     */
    fun getUserStats(): Flow<UserStats>

    /**
     * 获取所有跑步活动
     */
    fun getAllActivities(): Flow<List<RunningActivity>>

    /**
     * 获取最新活动
     */
    fun getLatestActivity(): Flow<RunningActivity?>

    /**
     * 获取所有路线
     */
    fun getAllRoutes(): Flow<List<Route>>

    /**
     * 根据ID获取路线
     */
    suspend fun getRouteById(routeId: String): Route?

    /**
     * 保存跑步活动
     */
    suspend fun saveActivity(activity: RunningActivity)

    /**
     * 同步数据
     */
    suspend fun syncData(): Result<Unit>
}

File: ios/Domain/Repositories/RunningRepository.swift
import Foundation
import Combine

/// 跑步数据 Repository 协议
protocol RunningRepository {
    /// 获取用户统计数据
    func getUserStats() -> AnyPublisher<UserStats, Never>

    /// 获取所有跑步活动
    func getAllActivities() -> AnyPublisher<[RunningActivity], Never>

    /// 获取最新活动
    func getLatestActivity() -> AnyPublisher<RunningActivity?, Never>

    /// 获取所有路线
    func getAllRoutes() -> AnyPublisher<[Route], Never>

    /// 根据ID获取路线
    func getRouteById(_ routeId: String) -> Route?

    /// 保存跑步活动
    func saveActivity(_ activity: RunningActivity)

    /// 同步数据
    func syncData() -> AnyPublisher<Void, Error>
}
